src/messages/embed.ts

```diff
@@ -0,0 +1,300 @@
+	 */
+	public addField(name: string, value: string, inline = false): this {
+		// Ensure adding this field won't exceed the 25 field limit
+		validateFieldLength(this.fields, 1);
+
+		return this.addFields({ name, value, inline });
+	}
+
+	/**
+	 * Adds fields to the embed (max 25).
+	 * @param fields The fields to add.
+	 */
+	public addFields(...fields: APIEmbedField[]): this {
+		// Data assertions
+		ow(fields, embedFieldsArrayPredicate);
+
+		// Ensure adding this field won't exceed the 25 field limit
+		validateFieldLength(this.fields, fields.length);
+
+		this.fields.push(...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, and inserts fields in the embed (max 25).
+	 * @param index The index to start at.
+	 * @param deleteCount The number of fields to remove.
+	 * @param fields The replacing field objects.
+	 */
+	public spliceFields(index: number, deleteCount: number, ...fields: APIEmbedField[]): this {
+		// Data assertions
+		ow(fields, 'fields', embedFieldsArrayPredicate);
+
+		// Ensure adding this field won't exceed the 25 field limit
+		validateFieldLength(this.fields, fields.length - deleteCount);
+
+		this.fields.splice(index, deleteCount, ...Embed.normalizeFields(...fields));
+		return this;
+	}
+
+	/**
+	 * Sets the author of this embed.
+	 * @param name The name of the author.
+	 * @param options The URL options of the author.
+	 * @param options.iconURL The icon URL of the author.
+	 * @param options.url The URL of the author.
+	 */
+	public setAuthor(name: string | null, { iconURL, url }: Record<string, string | undefined | null> = {}): this {
+		// Data assertions
+		ow(name, 'name', ow.any(ow.string.minLength(1).maxLength(256), ow.null));
+		ow(iconURL, 'iconURL', ow.any(ow.string.url, ow.nullOrUndefined));
+		ow(url, 'url', ow.any(ow.string.url, ow.nullOrUndefined));
+
+		this.author = name ? { name, icon_url: iconURL ?? undefined, url: url ?? undefined } : undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the color of this embed.
+	 * @param color The color of the embed.
+	 */
+	public setColor(color: number | null): this {
+		// Data assertions
+		ow(color, 'color', ow.any(ow.number, ow.null));
+
+		this.color = color ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the description of this embed.
+	 * @param description The description.
+	 */
+	public setDescription(description: string | null): this {
+		// Data assertions
+		ow(description, 'description', ow.any(ow.string.minLength(1).maxLength(4096), ow.null));
+
+		this.description = description ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the footer of this embed.
+	 * @param text The text of the footer.
+	 * @param iconURL The icon URL of the footer.
+	 */
+	public setFooter(text: string, iconURL?: string | null): this {
+		// Data assertions
+		ow(text, 'text', ow.any(ow.string.minLength(1).maxLength(2048), ow.null));
+		ow(iconURL, 'iconURL', ow.any(ow.string.url, ow.nullOrUndefined));
```

Comment:
Move the predicates to the top, as opposed to processing that object over and over.

Reply:
Honestly, the same goes for everything else too.